lib/instance.js

```diff
@@ -869,7 +896,19 @@ Instance.prototype.destroy = function(options) {
 
       this.setDataValue(field, values[field]);
 
-      return this.sequelize.getQueryInterface().update(this, this.$Model.getTableName(options), values, where, _.defaults({ hooks: false }, options));
+      return this.sequelize.getQueryInterface().update(
+        this, this.$Model.getTableName(options), values, where, _.defaults({ hooks: false }, options)
+      ).then(function(results) {
+        var rowsUpdated = results[1];
+        if (this.Model._versionAttribute && rowsUpdated < 1) {
+          throw new sequelizeErrors.OptimisticLockError({
+            model: this.Model.name,
+            values: values,
+            where: where,
+          });
+        }
+        return _.head(results);
```

Comment:
Hmm, I think that would be a much bigger change.  Looks like `Model.update` uses a `bulkInsert` query.